content/handling-multiple-k8sclusters.md

```diff
@@ -0,0 +1,69 @@
+
+## Kubernetes Configuration Files and Context Switching
+
+Before we discuss multiple clusters, let's take a brief look at how kubectl finds out how to talk to a particular Kubernetes cluster. By the way, if you're looking for a general introduction to kubectl, have a look at [this tutorial](https://rancher.com/learning-paths/how-to-manage-kubernetes-with-kubectl/) by Kelly Griffin. 
+
+The most widely used mechanism to find cluster access information are so-called _kubeconfig files_. By default, kubectl will use information from the file `$HOME/.kube/config`. It is possible to point to other config files by setting the `KUBECONFIG` environment variable or by using the `--kubeconfig` flag. 
+
+There are three main components in a kubeconfig file:
+* clusters: describes your - you guessed it - clusters, i.e. specifies name, API server URL and certificate authority. 
+* users: the user credentials needed to authenticate to your cluster(s), i.e. name and either key pairs or username/password pairs
+* context: a combination of cluster, user and namespace to exactly define the target for your subsequent kubectl operations. Even if you only have one cluster, you can either have multiple users or different namespaces to work with, so the context construct makes it easier to switch between your most frequently used combinations. 
+
+You can of course just manually edit your kubeconfig files, but you can also manipulate them with a number of `kubectl config` subcommands:
+* `kubectl config --kubeconfig=config-demo set-cluster` to define clusters
+* `kubectl config --kubeconfig=config-demo set-credentials` for setting user credentials
+* `kubectl config --kubeconfig=config-demo set-context` to define or change contexts
+
+Once you have set up your clusters, users and contexts, you can easily switch between contexts using the `kubectl config use-context` command. The currently active context can be checked with `kubectl config current-context`. 
+
+A more detailed description of all this can be found at [https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/]. 
+
+Theoretically, this could be the end of the article, if it wasn't for the fact that once the number of clusters you interact with grows beyond low single-digit numbers, keeping all information in one config file becomes, let's say, unpleasant. 
+
+## Multiple config files
+
+A better way (in my humble opinion) is to keep information about different clusters in different config files. There's basically two ways to handle multiple config files: the simple way and the clever way. The simple way is you manually specify the config you want to use each time you call kubectl via the `--kubeconfig` flag. The clever way is to concatenate multiple kubeconfigs into your `KUBECONFIG` environment variable like 
+
+    export KUBECONFIG=$HOME/.kube/config:second-config-file:third-config-file. 
```

Comment:
Have you tested this? I couldn't get it to work the last time I tried but maybe I was doing something wrong...

Reply:
I'm running this on my machine and it works fine.